### PR TITLE
feat: add extension point to the top of welcome page

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

- adds a registration point for adding components to the top of welcome page
- add new dark color and new icon

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/constants.ts [hunk 1]
```diff
@@ -95,6 +95,10 @@ export const URL_PARAMS = {
     name: 'viz_type',
     type: 'string',
   },
+  showDatabaseModal: {
```
Comment (suddjian) [inline superset-frontend/src/constants.ts:98]: why is this added?

File: superset-frontend/packages/superset-ui-core/src/style/index.tsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ const defaultTheme = {
       base: '#20A7C9',
       dark1: '#1A85A0',
       dark2: '#156378',
+      dark3: '#0C4250',
```
Comment (suddjian) [inline superset-frontend/packages/superset-ui-core/src/style/index.tsx:65]: why is this added?

File: superset-frontend/packages/superset-ui-core/src/ui-overrides/UiOverrideRegistry.ts [hunk 3]
```diff
@@ -35,6 +35,7 @@ export type UiOverrides = Partial<{
   'embedded.documentation.description': UiGeneratorText;
   'embedded.documentation.url': string;
   'navbar.right': React.ComponentType;
+  'welcome.top': React.ComponentType;
```
Comment (suddjian) [inline superset-frontend/packages/superset-ui-core/src/ui-overrides/UiOverrideRegistry.ts:38]: I think a good name for this would be `welcome-page.banner`

**Comment by suddjian** [comment]
I removed that asset since I believe that's meant to be part of the Preset feature, not part of Superset

File: superset-frontend/src/constants.ts (hunk 1 above)
Comment (lilykuang) [reply superset-frontend/src/constants.ts:98]: one of the setup guide step is to redirect to database list view and pop open the database modal

File: superset-frontend/packages/superset-ui-core/src/style/index.tsx (hunk 2 above)
Comment (lilykuang) [reply superset-frontend/packages/superset-ui-core/src/style/index.tsx:65]: i will remove this

**Review by suddjian — APPROVED** [review]
sweet
